specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-08-02-preview/managedClusters.json

```diff
@@ -3556,9 +3556,87 @@
         "clientId"
       ]
     },
+    "FlexProfile": {
+      "type": "object",
+      "properties": {
+        "scaleProfile": {
+          "$ref": "#/definitions/ScaleProfile",
+          "description": "Specifications on how to scale the AgentPool."
+        }
+      },
+      "description": "Specifications on flex Agentpool."
+    },
+    "ScaleProfile": {
+      "type": "object",
+      "properties": {
+        "manual": {
+          "$ref": "#/definitions/ManualScaleProfile",
+          "description": "Specifications on how to scale the flex agent pool to a fixed size."
+        }
+      },
+      "description": "Specifications on how to scale the AgentPool."
+    },
+    "ManualScaleProfile": {
+      "type": "object",
+      "properties": {
+        "machines": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManualScaleMachineSpec"
+          },
+          "description": "Specifications on how to scale the AgentPool."
+        }
+      }
+    },
+    "ManualScaleMachineSpec": {
+      "type": "object",
+      "properties": {
+        "sizes": {
+          "items": {
+            "type": "string"
+          },
+          "description": "The comma-separated list allowed vm sizes. AKS will use the first available one upon scale."
```

Comment:
Why comma-separated list versus an actual JSON list?

Reply:
We want to avoid ordering confusion on lists during parsing? I remember some corner cases there.

Reply:
If this is azure api recommendation we can use list - the two goals here are:
1. need to make sure we keep the ordering.
2. make sure cx can updating the list(yes comma seperated string or array we'll always have to ask cx to patch the full list )

Reply:
If it's a list they will have to patch the full list as well (no list merge). If patching a single item is required itd need to be a map of some sort, but I sorta suspect we dont want/need individual item patching